ui/components/app/menu-bar/account-options-menu.js

```diff
@@ -67,9 +78,12 @@ export default function AccountOptionsMenu({ anchorElement, onClose }) {
         }
         iconClassName="fas fa-external-link-alt"
       >
-        {rpcPrefs.blockExplorerUrl
-          ? t('viewinExplorer', [t('blockExplorerAccountAction')])
-          : t('viewOnEtherscan', [t('blockExplorerAccountAction')])}
+        {rpcPrefs.blockExplorerUrl &&
+          t('viewinExplorer', [t('blockExplorerAccountAction')])}
+        {!rpcPrefs.blockExplorerUrl && isCustomNetwork && t('addBlockExplorer')}
+        {!rpcPrefs.blockExplorerUrl &&
+          !isCustomNetwork &&
+          t('viewOnEtherscan', [t('blockExplorerAccountAction')])}
```

Comment:
This is kind of hard to follow the logic, my suggestion would be to move this outside of the JSX return and instead do something like:

```js

const routeToAddBlockExplorerUrl = () => {
  history.push(`${NETWORKS_ROUTE}#blockExplorerUrl`);
};

const openBlockExplorer = () => {
  trackEvent({
    event: 'Clicked Block Explorer Link',
    category: 'Navigation',
    properties: {
      link_type: 'Account Tracker',
      action: 'Account Options',
      block_explorer_domain: getURLHostName(addressLink),
    },
  });
  global.platform.openTab({
    url: addressLink,
  });
  onClose();
};

// If the user is on a custom network and doesn't have a block explorer set in state the text
// will display "Add block explorer" and route the user to the network settings page to add
// a block explorer. If a block explorer url has been added and is known a new  tab will be
// opened to that block explorer and the text will say "view in explorer". Finally, the block
// explorer will not be set in the case of the MetaMask default networks, and the text will
// indicate that etherscan is the explorer and the link will open a new tab to etherscan.
let blockExplorerLinkText = t('addBlockExplorer');
let blockExplorerOnClick = routeToAddBlockExplorerUrl;
if (rpcPrefs.blockExplorerUrl) {
  blockExplorerLinkText = t('viewInExplorer', [t('blockExplorerAccountAction')]);
  blockExplorerOnClick = openBlockExplorer;
} else if (isCustomNetwork === false) {
  blockExplorerLinktext = t('viewOnEtherscan', [t('blockExplorerAccountAction')]);
  blockExplorerOnClick = openBlockExplorer;
}
```

disclaimer: This is pseudo-pseudo code written in browser, please review it, apply the *concept* to your work if you agree. :)

Reply:
PS: IT seems that this logic is done in many places. It might be beneficial to change this into a redux selector, action or a react hook (please use your best judgement) to prevent duplication of this logic as much as possible.

Reply:
@brad-decker 
I did what you had suggested using the redux selector in this case. The new logic can be seen here: a40ddd64347037c7fc7ddee838263712cd149c92
Hope I got the most out of it. :) 
Also, these are the options that were considered during the code refactoring:

- addBlockExplorer
- viewinExplorer
- viewOnEtherscan 
- viewOnBlockExplorer
- blockExplorerView
- etherscanViewOn
- etherscanView